Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -123,13 +128,29 @@ export class azureclitask {
 
         var tenantId: string = tl.getEndpointAuthorizationParameter(connectedService, "tenantid", false);
         var subscriptionID: string = tl.getEndpointDataParameter(connectedService, "SubscriptionID", true);
+
+        // set az cli config dir
+        this.setConfigDirectory();
+
         //login using svn
         this.throwIfError(tl.execSync("az", "login --service-principal -u \"" + servicePrincipalId + "\" -p \"" + cliPassword + "\" --tenant \"" + tenantId + "\""), tl.loc("LoginFailed"));
         this.isLoggedIn = true;
         //set the subscription imported to the current subscription
         this.throwIfError(tl.execSync("az", "account set --subscription \"" + subscriptionID + "\""), tl.loc("ErrorInSettingUpSubscription"));
     }
 
+    private static setConfigDirectory(): void {
+        var configDirName: string = "c" + new Date().getTime(); // 'c' denotes config
+        if (tl.osType().match(/^Win/)) {
+            this.azCliConfigPath = path.join(process.env.USERPROFILE, ".azclitask", configDirName);
+        }
+        else {
+            this.azCliConfigPath = path.join(process.env.HOME, ".azclitask", configDirName);
```

Comment:
Are we sure that we always have permission to create new folder permissions in these paths ? 
Should we use some agent folder only or temp folder of respective os ?